spotify_player/src/state/model.rs

```diff
@@ -55,7 +56,42 @@ pub enum ContextId {
 /// An offset can be either a track's URI or its absolute offset in the context
 pub enum Playback {
     Context(ContextId, Option<rspotify_model::Offset>),
-    URIs(Vec<TrackId<'static>>, Option<rspotify_model::Offset>),
+    URIs(Vec<PlayableId<'static>>, Option<rspotify_model::Offset>),
+}
+
+impl std::fmt::Debug for Playback {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Self::Context(context_id, offset) => f
+                .debug_tuple("Playback::Context")
+                .field(context_id)
+                .field(offset)
+                .finish(),
+            Self::URIs(playable_ids, offset) => {
+                write!(f, "Playback::URIs([")?;
+                for id in playable_ids.iter() {
+                    match id {
+                        PlayableId::Track(track_id) => write!(f, "{}, ", track_id)?,
+                        PlayableId::Episode(episode_id) => write!(f, "{}, ", episode_id)?,
+                    }
+                }
+                write!(f, "], ")?;
+                write!(f, "{:?})", offset)
+            }
+        }
+    }
+}
+
+impl Clone for Playback {
+    fn clone(&self) -> Self {
+        match self {
+            Self::Context(context_id, offset) => Self::Context(context_id.clone(), offset.clone()),
+            Self::URIs(playable_ids, offset) => Self::URIs(
+                playable_ids.into_iter().map(|x| x.clone_static()).collect(),
```

Comment:
There's a lot of painful places like this throughout the diff where things would be easier if `PlayableId` had `impl Clone`. Will open a PR against `rspotify` to see if that could be added

Reply:
https://github.com/ramsayleung/rspotify/pull/490